onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -229,8 +229,10 @@ class PlannerImpl {
   int& UseCount(const OrtValueName& name) { return UseCount(Index(name)); }
 
   int DecrementUseCount(OrtValueIndex n) {
-    int& use_count = --UseCount(n);
-    assert(use_count >= 0);
+    int& use_count = UseCount(n);
+    if (use_count > 0) {
+      --use_count;
+    }
```

Comment:
This is an invariant that should never be violated. Therefore, the assert was appropriate in debug builds. The assert should stay to catch a bug. If this assert somehow fires, we need to investigate it, not remove.
If we want something in production, should we at least log a warning if not ORT_ENFORCE?

Reply:
If usage is like if (UseCount(n) >0) DecrementUseCount(n), we can add assert or ORT_ENFORCE that use_count will not be negative. But the usage does not check use_count before decreasing, I do not think we shall put assert here.